emcc.py

```diff
@@ -3169,6 +3169,8 @@ def consume_arg_file():
         else:
           # for 3+, report -g3 to clang as -g4 etc. are not accepted
           newargs[i] = '-g3'
+          if settings.DEBUG_LEVEL == 3:
+            settings.GENERATE_DWARF = 1
```

Comment:
Now that this exists, maybe a nice followup PR could use it in more places. I think we have things like `DEBUG_LEVEL == 3` in various places.

Reply:
Good idea, will check for that.

Reply:
Tried to search for the usage, but it's tricky. Usually when we check for `DEBUG_LEVEL == 3` or `DEBUG_LEVEL >= 3`, it's for generating/preserving DWARF info. But the tricky thing is, even if only source map is required, it needs DWARF info to be generated until the middle of the process and we only get rid of DWARF at the end after generating source map from it. So `DEBUG_LEVEL == 3` is synonymous with "Do we need DWARF info _in the beginning_?", and I'm not sure if replacing this with `if settings.GENERATE_DWARF or settings.GENERATE_SOURCE_MAP` will be much of a improvement. If we want to go that route, I think it's better to redefine the meaning of `DEBUG_LEVEL` or create separate variables for other `DEBUG_LEVEL` (1 and 2) too; but I'm not sure if we want to go that far.

Reply:
Makes sense, thanks for checking. Nothing to do for that, then.